cladldevice.go

```diff
@@ -209,12 +259,19 @@ func NewDevice(index int, order int, platformID cl.CL_platform_id, deviceID cl.C
 		platformID:  platformID,
 		deviceID:    deviceID,
 		deviceName:  getDeviceInfo(deviceID, cl.CL_DEVICE_NAME, "CL_DEVICE_NAME"),
-		kind:        "adl",
+		deviceType:  getDeviceInfo(deviceID, cl.CL_DEVICE_TYPE, "CL_DEVICE_TYPE"),
 		quit:        make(chan struct{}),
 		newWork:     make(chan *work.Work, 5),
 		workDone:    workDone,
 		fanPercent:  0,
 		temperature: 0,
+		tempTarget:  0,
+	}
+
+	if d.deviceType == "CPU" {
+		d.kind = DeviceKindUnknown
```

Comment:
We were using determineDeviceKind() before the build was broken up into 3. Now we hard code the types in the cuda and opencladl portions.